src/pke/lib/scheme/bfvrns/bfvrns-pke.cpp

```diff
@@ -68,14 +125,49 @@ Ciphertext<DCRTPoly> PKEBFVRNS::Encrypt(DCRTPoly ptxt,
           privateKey->GetCryptoParameters());
 
   const auto elementParams = cryptoParams->GetElementParams();
+  auto encParams = elementParams;
+
+  if (cryptoParams->GetEncryptionTechnique() == POVERQ) {
+    encParams = cryptoParams->GetParamsQr();
+    ptxt.SetFormat(Format::COEFFICIENT);
+    Poly bigPtxt = ptxt.CRTInterpolate();
+    DCRTPoly plain(bigPtxt, encParams);
+    ptxt = plain;
+    ptxt.SetFormat(Format::COEFFICIENT);
+  }
 
   std::shared_ptr<std::vector<DCRTPoly>> ba =
-      EncryptZeroCore(privateKey, elementParams);
+      EncryptZeroCore(privateKey, encParams);
 
-  ptxt.SetFormat(Format::EVALUATION);
+  if (cryptoParams->GetEncryptionTechnique() == POVERQ) {
+    const NativeInteger &NegQrModt = cryptoParams->GetNegQrModt();
+    const NativeInteger &NegQrModtPrecon = cryptoParams->GetNegQrModtPrecon();
+    const std::vector<NativeInteger> &tInvModq = cryptoParams->GettInvModq();
+    const std::vector<NativeInteger> &tInvModqPrecon = cryptoParams->GettInvModqPrecon();
+    const NativeInteger t = cryptoParams->GetPlaintextModulus();
+
+    DCRTPoly prod = ptxt;
+    prod.TimesQovert(encParams, tInvModq, tInvModqPrecon, t, NegQrModt, NegQrModtPrecon);
+    prod.SetFormat(Format::EVALUATION);
```

Comment:
can you pull 2 lines out of this if statement and add them after?

Reply:
Discussed with Dmitriy.